### ISSUE TITLE
train

### ISSUE THREAD

**Issue by huangmengxiao000** [erprogs/CViT#13]
你好，请问这里的xm和pl是什么？
for inputs, labels in pl.ParallelLoader(dataloaders[phase], [device]).per_device_loader(device):
                inputs = inputs.to(device)
                labels = labels.to(device)
if phase == 'train':
                        loss.backward()
                        xm.optimizer_step(optimizer, barrier=True) 
                        xm.mark_step()

**Comment by erprogs** [comment]
Hello @huangmengxiao000 
pl is an alias for distributed parallel loader for training on TPU. Its is defined in [loader ](https://github.com/erprogs/CViT/blob/main/helpers/loader.py) file, xm is torch model for TPU.
`import torch_xla.distributed.parallel_loader as pl`
`import torch_xla.core.xla_model as xm`
